samples/outTreeEp/out_tree_ep.cc

```diff
@@ -0,0 +1,105 @@
+#include "out_tree_ep.h"
+#include <memory>
+#include <vector>
+namespace onnxruntime {
+
+OutTreeEp::OutTreeEp(const char* ep_type, const OutTreeEpInfo& ep_info) : info(ep_info) {
+    type = ep_type;
+    OrtExecutionProvider::GetCapability = [](const OrtExecutionProvider* this_, const OrtGraphViewer* graph, size_t* cnt, OrtIndexedSubGraph*** indexed_sub_graph) {
+        const OrtApi* api = OrtGetApiBase()->GetApi(ORT_API_VERSION);
+        std::vector<OrtIndexedSubGraph*> cache;
+        size_t nodes_count = 0;
+        const size_t* nodes_index = nullptr;
+        api->OrtGraph_GetNodesIndexInTopologicalOrder(graph, &nodes_count, &nodes_index);
+        for (size_t i = 0; i < nodes_count; i++) {
+            const OrtNode* node = nullptr;
+            api->OrtGraph_GetOrtNode(graph, nodes_index[i], &node);
+            const char* node_op_type;
+            api->OrtNode_GetOpType(node, &node_op_type);
+            if (!strcmp(node_op_type, "Relu")) {
+                OrtIndexedSubGraph* subgraph = new OrtIndexedSubGraph();
+                subgraph->node_index_len = 1;
+                subgraph->node_index = new size_t [subgraph->node_index_len];
+                subgraph->node_index[0] = nodes_index[0];
+
+                subgraph->meta_def = new OrtMetaDef();
+                subgraph->meta_def->name = "Relu_subgraph";
+                subgraph->meta_def->input_len = 0;
+                api->OrtNode_GetInputSize(node, &(subgraph->meta_def->input_len));
+                subgraph->meta_def->inputs = new const char* [subgraph->meta_def->input_len];
+                for (size_t j = 0; j < subgraph->meta_def->input_len; j++) {
+                    subgraph->meta_def->inputs[j] = nullptr;
+                    api->OrtNode_GetIthInputName(node, j, &(subgraph->meta_def->inputs[j]));
+                }
+
+                subgraph->meta_def->output_len = 0;
+                api->OrtNode_GetOutputSize(node, &(subgraph->meta_def->output_len));
+                subgraph->meta_def->outputs = new const char* [subgraph->meta_def->output_len];
+                for (size_t j = 0; j < subgraph->meta_def->output_len; j++) {
+                    subgraph->meta_def->outputs[j] = nullptr;
+                    api->OrtNode_GetIthOutputName(node, j, &(subgraph->meta_def->outputs[j]));
+                }
+
+                cache.push_back(subgraph);
+            }
+        }
+
+        *cnt = cache.size();
+        *indexed_sub_graph = new OrtIndexedSubGraph* [*cnt];
+        for (size_t i = 0; i < *cnt; i++) {
+            (*indexed_sub_graph)[i] = cache[i];
+        }
+    };
+
+    OrtExecutionProvider::Compile = [](OrtExecutionProvider* this_, const OrtGraphViewer** graph, const OrtNode** node, size_t cnt, OrtNodeComputeInfo*** node_compute_info) {
+        for (size_t i = 0; i < cnt; i++) {
+            (*node_compute_info)[i] = new OrtNodeComputeInfo();
```

Comment:
I think allocation/memory handling of these OrtNodeComputeInfo objects also needs to be adjusted.

It looks like ORT is allocating the array of _pointers_ to OrtNodeComputeInfo objects, passing it to the EP, and then the EP allocates memory for each individual OrtNodeComputeInfo object (which are also leaked).

Instead, it may be simpler to have ORT allocate an array of OrtNodeComputeInfo objects (not pointers to them), and then have the EP fill them out without having to allocate any memory.

Example:
```c++
// In ORT code
node_compute_info_ = std::vector<OrtNodeComputeInfo>(count);
ep_impl_->Compile(/*...*/, count, node_compute_info_.data());

// In EP code
 OrtExecutionProvider::Compile = [](/*...*/, size_t cnt, OrtNodeComputeInfo** node_compute_info) {
    node_compute_info[i]->ComputeFunc = // initialize. No need to allocate any memory
}
```